plugins/transformations/json/srcC/FlatTypeTransformation.c

```diff
@@ -632,82 +111,47 @@ static DDS_ReturnCode_t
 {
     DDS_ReturnCode_t retcode = DDS_RETCODE_ERROR;
     DDS_ExceptionCode_t ex = DDS_NO_EXCEPTION_CODE;
-    DDS_UnsignedLong buffer_member_id = 0;
     struct DDS_TypeCode *member_type = NULL, *member_content_type = NULL;
     DDS_TCKind tckind = DDS_TK_NULL;
 
     RTI_TSFM_LOG_FN(RTI_TSFM_Json_FlatTypeTransformation_validate_buffer_member)
 
-    /* Currently we don't validate nor create a mapping for the "buffer member"
-       because users specify it as a possibly nested selector but
-       DDS_TypeCode_find_member_by_name doesn't support nested selectors. */
-
-#if 0
-
     /* Check that output type has the specified buffer member */
-    buffer_member_id = 
-        DDS_TypeCode_find_member_by_name(base_type, member_name, &ex);
-    if (DDS_NO_EXCEPTION_CODE != ex)
-    {
-        /* TODO Log error */
-        goto done;
-    }
-    member_type = DDS_TypeCode_member_type(base_type, buffer_member_id, &ex);
-    if (ex != DDS_NO_EXCEPTION_CODE)
-    {
-        /* TODO Log error */
-        goto done;
-    }
-    tckind = DDS_TypeCode_kind(member_type, &ex);
-    if (ex != DDS_NO_EXCEPTION_CODE)
-    {
-        /* TODO Log error */
+    member_type = RTI_COMMON_TypeCode_get_member_type(base_type, member_name);
+    if (member_type == NULL) {
         goto done;
     }
 
-    if (tckind != DDS_TK_SEQUENCE)
-    {
+    tckind = DDS_TypeCode_kind(member_type, &ex);
+    if (ex != DDS_NO_EXCEPTION_CODE) {
         /* TODO Log error */
         goto done;
     }
 
-    member_content_type = DDS_TypeCode_content_type(member_type, &ex);
-    if (ex != DDS_NO_EXCEPTION_CODE)
-    {
-        /* TODO Log error */
-        goto done;
-    }
+    /* In case of a Sequence, check it is a DDS_OctetSeq */
+    if (tckind == DDS_TK_SEQUENCE) {
+        member_content_type = DDS_TypeCode_content_type(member_type, &ex);
+        if (ex != DDS_NO_EXCEPTION_CODE) {
+            /* TODO Log error */
+            goto done;
+        }
 
-    tckind = DDS_TypeCode_kind(member_content_type, &ex);
-    if (ex != DDS_NO_EXCEPTION_CODE)
-    {
-        /* TODO Log error */
-        goto done;
-    }
+        tckind = DDS_TypeCode_kind(member_content_type, &ex);
+        if (ex != DDS_NO_EXCEPTION_CODE) {
+            /* TODO Log error */
+            goto done;
+        }
 
-    if (tckind != DDS_TK_OCTET)
-    {
+        if (tckind != DDS_TK_OCTET) {
```

Comment:
Since we're adding support for string members, I wonder if we should also add support for other types which might contain a string:
- `sequence<char>`
- `octet[]`
- `char[]`

Reply:
I've added some feedback from the PR, but this is still missing